server/src/main/java/org/apache/cassandra/sidecar/routes/cdc/ListCdcDirHandler.java

```diff
@@ -102,7 +109,7 @@ protected void handleInternal(RoutingContext context,
             LOGGER.warn("Error listing the CDC commit log segments", cause);
             context.response()
                    .setStatusCode(HttpResponseStatus.INTERNAL_SERVER_ERROR.code())
-                   .setStatusMessage(cause.getMessage())
+                   .setStatusMessage(cause != null ? cause.getMessage() : "Error while listing CDC segments")
```

Comment:
Do you see response message with 'null'?

Reply:
Yes I was seeing NPE

Reply:
How can the `cause` be null in the failure handler? I guess it is `cause.getMessage() == null` instead? If so, the change here does not make a difference. Probably get a better understanding of the issue first.
It is also unrelated with the patch. Given the change is tiny, I am fine.

Reply:
Sorry meant to do this `cause.getMessage() == null ? "Error while listing CDC segments" : cause.getMessage()` `cause.getMessage()` could be null and `setStatusMessage` does not allow setting null, throws NPE

Reply:
Sounds good. Can you have this instead? It is a nit.

```suggestion
                   .setStatusMessage(Objects.requireNonNullElse(cause.getMessage(), "Error while listing CDC segments"))
```